diff --git a/README-ubuntu b/README-ubuntu
index 3f0c8f9..b091a35 100644
--- a/README-ubuntu
+++ b/README-ubuntu
@@ -15,12 +15,6 @@
         libgmp3-dev libmpc-dev libmpfr-dev  \
         gcc-multilib
 
-    curl http://www.fit.vutbr.cz/~idudka/download/predator-2011-10-10-e280e28.tar.gz \
-        | gzip -cd \
-        | tar x
-
-    cd predator-2011-10-10-e280e28
-
     make -j3 build_boost build_gcc
 
     make -j3
