diff -rup invader-1_1.orig/sources/error.ml invader-1_1/sources/error.ml
--- invader-1_1.orig/sources/error.ml	2008-12-18 16:28:21.000000000 +0100
+++ invader-1_1/sources/error.ml	2009-11-01 01:49:31.444581940 +0100
@@ -67,10 +67,7 @@ let set_print_colour c = match c with
 (** Can be applied to any number of arguments and throws them all away *)
 let rec throw_away x = Obj.magic throw_away
 
-let log fmt =
-  if !Config.test
-  then Obj.magic throw_away
-  else Format.fprintf Format.std_formatter fmt
+let log fmt = Format.fprintf Format.std_formatter fmt
 
 let err fmt =
   let () = if !Config.num_cores > 1 then
