From a10e0091f616359f664a9782f87e59c36c35ebdb Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Sun, 1 Nov 2009 11:57:24 +0100
Subject: [PATCH 4/6] error: force Invader to be more verbose

---
 sources/error.ml |    5 +----
 1 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/sources/error.ml b/sources/error.ml
index 77153fe..ce126e7 100644
--- a/sources/error.ml
+++ b/sources/error.ml
@@ -67,10 +67,7 @@ let set_print_colour c = match c with
 (** Can be applied to any number of arguments and throws them all away *)
 let rec throw_away x = Obj.magic throw_away
 
-let log fmt =
-  if !Config.test
-  then Obj.magic throw_away
-  else Format.fprintf Format.std_formatter fmt
+let log fmt = Format.fprintf Format.std_formatter fmt
 
 let err fmt =
   let () = if !Config.num_cores > 1 then
-- 
1.6.5.rc1

