From ef240b3ebd3e007978a2b8c841d7e154199a1c4a Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Sun, 1 Nov 2009 14:46:58 +0100
Subject: [PATCH] sil: tweak colors

---
 sources/sil.ml |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/sources/sil.ml b/sources/sil.ml
index b49afd0..beaad05 100644
--- a/sources/sil.ml
+++ b/sources/sil.ml
@@ -633,12 +633,12 @@ let pp_exp_typ f (e,t) =
 
 (** Pretty print an instruction. *)
 let pp_instr f = function
-  | Letderef (id,e,t,loc) -> fprintf f "%a=*%a:%a %a" pp_ident id pp_exp e pp_typ t pp_loc loc
-  | Set (e1,t,e2,loc) -> fprintf f "*%a:%a=%a %a" pp_exp e1 pp_typ t pp_exp e2 pp_loc loc
+  | Letderef (id,e,t,loc) -> fprintf f "%a \x1B[1;33m=\x1B[0m \x1B[1;33m*\x1B[0m%a:%a %a" pp_ident id pp_exp e pp_typ t pp_loc loc
+  | Set (e1,t,e2,loc) -> fprintf f "\x1B[1;33m*\x1B[0m%a \x1B[1;33m:%a=\x1B[0m %a %a" pp_exp e1 pp_typ t pp_exp e2 pp_loc loc
   | Call (eto,e,arg_ts,loc) ->
       (match eto with
 	| None -> ()
-	| Some (e1,t1,root_t) -> fprintf f "*%a=(%a)" pp_exp e1 pp_typ t1);
+	| Some (e1,t1,root_t) -> fprintf f "*%a = (%a)" pp_exp e1 pp_typ t1);
       fprintf f "%a(%a) %a" pp_exp e (pp_seq pp_exp_typ) arg_ts pp_loc loc
 
 let rec pp_instr_list f = function
@@ -692,7 +692,7 @@ and pp_hpara f pred =
 
 and pp_dll_hpara f pred =
   let (iF,oB,oF,svars, evars,b) = (pred.cell, pred.blink,pred.flink, pred.svars_dll, pred.evars_dll, pred.body_dll) 
-  in fprintf f "lam [%a,%a,%a,%a]. exists [%a]. %a" 
+  in fprintf f "\x1B[1;31mlam [\x1B[0m%a\x1B[1;31m,\x1B[0m%a\x1B[1;31m,\x1B[0m%a\x1B[1;31m,\x1B[0m%a\x1B[1;31m]\x1B[0m. \x1B[1;31mex\x1B[0mists \x1B[1;31m[\x1B[0m%a\x1B[1;31m]\x1B[0m. %a" 
        pp_ident iF      
        pp_ident oB
        pp_ident oF      
-- 
1.6.5.rc1

