From 345e9c4fdd99a07b71f975580ddca1d6f1ffc697 Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Sun, 1 Nov 2009 18:31:48 +0100
Subject: [PATCH] Makefile: add target 'browse'

---
 sources/Makefile |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/sources/Makefile b/sources/Makefile
index 4f44b19..2eb5716 100644
--- a/sources/Makefile
+++ b/sources/Makefile
@@ -1,6 +1,9 @@
 tmp:
 	$(MAKE) -C cil
 
+browse:
+	ocamlbrowser -I cil/obj/x86_LINUX -w ""
+
 export OCAMLMAKEFILE = OcamlMakefile
 export OCAMLYACC = ocamlyacc -v
 export OCAMLC = ocamlc.opt -dtypes
-- 
1.6.5.rc1

