diff --git a/cl/config_cl.h b/cl/config_cl.h
index f2eaa0c..a98ad83 100644
--- a/cl/config_cl.h
+++ b/cl/config_cl.h
@@ -66,7 +66,7 @@
 /**
  * if 1, show the amount of time taken by the analysis even without verbose mode
  */
-#define CL_EASY_TIMER                   1
+#define CL_EASY_TIMER                   0
 
 /**
  * if 1, filter out repeated error/warning messages (sort of 2>&1 | uniq)
@@ -81,7 +81,7 @@
 /**
  * if 1, suppress warnings about unhandled code constructs
  */
-#define CLPLUG_SILENT                   0
+#define CLPLUG_SILENT                   1
 
 /**
  * if 1, print progress of code_listener factory when running in verbose mode
