diff --git a/switch-host-gcc.sh b/switch-host-gcc.sh
index b81ee1d..238b3eb 100755
--- a/switch-host-gcc.sh
+++ b/switch-host-gcc.sh
@@ -11,10 +11,10 @@ usage() {
     printf "Usage: %s GCC_HOST\n" "$SELF" >&2
     cat >&2 << EOF
 
-    Use this script to (re)build Predator and/or Forester against an arbitrary
-    build of host GCC.  The host GCC needs to be built with the support for GCC
-    plug-ins.  The currently supported version of host GCC is 4.7.0, but feel
-    free to use any other version of GCC at your own responsibility.
+    Use this script to (re)build Code Listener against an arbitrary build of
+    host GCC.  The host GCC needs to be built with the support for GCC plug-ins.
+    The currently supported version of host GCC is 4.7.0, but feel free to use
+    any other version of GCC at your own responsibility.
 
     GCC_HOST is the absolute path to gcc(1) that is built with the support for
     GCC plug-ins.  The most common location of the system GCC is /usr/bin/gcc.
@@ -70,5 +70,3 @@ build_analyzer() {
 }
 
 build_analyzer fwnull fwnull
-build_analyzer sl Predator
-build_analyzer fa Forester
