 tests/forester-regre/test-p0028.err | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/forester-regre/test-p0028.err b/tests/forester-regre/test-p0028.err
index f31b844..8cb6bf3 100644
--- a/tests/forester-regre/test-p0028.err
+++ b/tests/forester-regre/test-p0028.err
@@ -1,2 +1,2 @@
-test-p0028.c:24: note: item = malloc ((int)4U)
+test-p0028.c:24: note: item = _
 test-p0028.c:24: error: allocated block size mismatch
