diff --git a/docs/README-ubuntu b/docs/README-ubuntu
index e8dc24d..904618a 100644
--- a/docs/README-ubuntu
+++ b/docs/README-ubuntu
@@ -6,12 +6,6 @@ The below steps were successfully tested on a fresh installation of Ubuntu
 sudo apt-get install \
     cmake curl g++ gcc-4.6-multilib gcc-4.6-plugin-dev libboost-dev make
 
-curl http://www.fit.vutbr.cz/~idudka/download/predator-2013-02-02-8d3ea3d.tar.gz \
-    | gzip -cd \
-    | tar x
-
-cd predator-2013-02-02-8d3ea3d
-
 patch -p1 < build-aux/gcc-4.6.3.patch
 
 ./switch-host-gcc.sh /usr/bin/gcc-4.6
