diff --git a/docs/README-fedora b/docs/README-fedora
index fae4c88..7d89179 100644
--- a/docs/README-fedora
+++ b/docs/README-fedora
@@ -5,10 +5,4 @@ The below steps were successfully tested on a fresh installation of Fedora 17:
 sudo yum install \
     boost-devel cmake gcc-c++ glibc-devel glibc-devel.i686 gcc-plugin-devel make
 
-curl http://www.fit.vutbr.cz/~idudka/download/predator-2013-02-02-8d3ea3d.tar.gz \
-    | gzip -cd \
-    | tar x
-
-cd predator-2013-02-02-8d3ea3d
-
 ./switch-host-gcc.sh /usr/bin/gcc
