diff --git a/docs/README-ubuntu b/docs/README-ubuntu
index e8dc24d..904618a 100644
--- a/docs/README-ubuntu
+++ b/docs/README-ubuntu
@@ -15,12 +15,6 @@ This part is common for all versions of Ubuntu we have tried:
         libgmp3-dev libmpc-dev libmpfr-dev  \
         gcc-multilib
 
-    curl http://www.fit.vutbr.cz/~idudka/download/predator-2011-10-11-8a7846b.tar.gz \
-        | gzip -cd \
-        | tar x
-
-    cd predator-2011-10-11-8a7846b
-
     make -j3 build_boost build_gcc
 
     make -j3
