diff --git a/switch-host-gcc.sh b/switch-host-gcc.sh
index 071f50e..9cd8aba 100755
--- a/switch-host-gcc.sh
+++ b/switch-host-gcc.sh
@@ -11,10 +11,10 @@ usage() {
     printf "Usage: %s HOST_GCC_EXECUTABLE GCC_PLUGIN_INCLUDE_DIR\n" "$SELF" >&2
     cat >&2 << EOF
 
-    Use this script to (re)build Predator against an arbitrary build of host
-    GCC.  The host GCC needs to be built with the support for GCC plug-ins.  The
-    currently supported version of host GCC is 4.6.1, but feel free to use any
-    other version of GCC at your own responsibility.
+    Use this script to (re)build Code Listener against an arbitrary build of
+    host GCC.  The host GCC needs to be built with the support for GCC plug-ins.
+    The currently supported version of host GCC is 4.6.1, but feel free to use
+    any other version of GCC at your own responsibility.
 
     HOST_GCC_EXECUTABLE is an exectuable of gcc(1).  It can be given either with
     absolute path (e.g. /home/bob/gcc-4.7.0/bin/gcc) or, if it can be reached
@@ -73,17 +73,6 @@ status_update "Checking whether Code Listener works"
 make -C cl check \
     || die "Code Listener does not work"
 
-status_update "Trying to build Predator"
-make -C sl CMAKE="cmake -D GCC_HOST='$GCC_HOST'" \
-    || die "failed to build Predator"
-
-status_update "Checking whether Predator works"
-make -C sl check
-
-if test -d .git; then
-    die "you are in a git repository --> you need set \$GCC_HOST yourself!"
-else
-    status_update "Initializing \$GCC_HOST"
-    sed "s|GCC_HOST=.*\$|GCC_HOST='$GCC_HOST'|" -i \
-        chk-error-label-reachability.sh register-paths.sh sl/probe.sh sl/slgcc
-fi
+status_update "Trying to build fwnull"
+make -C fwnull CMAKE="cmake -D GCC_HOST='$GCC_HOST'" \
+    || die "failed to build fwnull"
