From f174a7bf619676bf0e0ac861951d68fbf9745ebc Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Sun, 19 Dec 2010 11:50:07 +0100
Subject: [PATCH 2/3] cdrom: comment out some code that causes us problems

---
 examples/drivers/cdrom.c |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/examples/drivers/cdrom.c b/examples/drivers/cdrom.c
index 8e3cbb0..5ee7b95 100644
--- a/examples/drivers/cdrom.c
+++ b/examples/drivers/cdrom.c
@@ -2724,6 +2724,7 @@ int register_cdrom(struct cdrom_device_info *cdi )
   banner_printed = (char)1;
   cdrom_sysctl_register();
  }
+#if 0
  if ((unsigned int )cdo->drive_status == (unsigned int )((void *)0)) {
   (*change_capability) &= -2049;
  }
@@ -2757,6 +2758,7 @@ int register_cdrom(struct cdrom_device_info *cdi )
  if ((unsigned int )cdo->generic_packet == (unsigned int )((void *)0)) {
   (*change_capability) &= -4097;
  }
+#endif
  cdi->mc_flags = 0U;
  cdo->n_minors = 0;
  cdi->options = 4;
@@ -3029,6 +3031,7 @@ static void cdrom_count_tracks(struct cdrom_device_info *cdi , tracktype *tracks
   return;
  }
 
+#if 0
  ret = get_nondet_int();
  if (ret) {
   if (ret == -123) {
@@ -3066,6 +3069,7 @@ static void cdrom_count_tracks(struct cdrom_device_info *cdi , tracktype *tracks
   }
   i ++;
  }
+#endif
  if (debug == 1) {
   HsPrintk("<6>cdrom: disc has %d tracks: %d=audio %d=data %d=Cd-I %d=XA\n", header.cdth_trk1,
       tracks->audio, tracks->data, tracks->cdi, tracks->xa);
@@ -6161,7 +6165,7 @@ int main_sub(void)
   return(tmp);
  }
 
- while ((get_nondet_int() > 0)) {
+ do {
   if ((get_nondet_int() > 0)) {
    med = get_nondet_ptr();
    tmp = cdrom_get_media_event(cdi, med);
@@ -6202,7 +6206,7 @@ int main_sub(void)
   else if ((get_nondet_int() > 0)) {
    tmp = cdrom_media_changed(cdi);
   }
- }
+ } while (0);
 
  tmp = unregister_cdrom(cdi);
  if (tmp == 0) {
-- 
1.7.3.2

